extension/js/common/api/email-provider/gmail/google-auth.ts

```diff
@@ -200,72 +189,39 @@ export class GoogleAuth {
     return await GoogleAuth.newAuthPopup({ acctEmail, scopes: GoogleAuth.defaultScopes('openid'), save: false });
   };
 
-  private static waitForOauthWindowClosed = async (oauthWinId: number, acctEmail: string | undefined): Promise<AuthRes> => {
-    return await new Promise(resolve => {
-      const onOauthWinClosed = (closedWinId: number) => {
-        if (closedWinId === oauthWinId) {
-          chrome.windows.onRemoved.removeListener(onOauthWinClosed);
-          resolve({ result: 'Closed', acctEmail, id_token: undefined });
-        }
-      };
-      chrome.windows.onRemoved.addListener(onOauthWinClosed);
-    });
+  private static getParameterByName = (search: string, name: string) => {
+    const match = RegExp('[?&]' + name + '=([^&]*)').exec(search);
+    return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
   };
 
-  private static processOauthResTitle = (title: string): { result: GoogleAuthWindowResult$result, code?: string, error?: string, csrf?: string } => {
-    const parts = title.split(' ', 2);
-    const result = parts[0] as GoogleAuthWindowResult$result;
-    const params = Url.parse(['code', 'state', 'error'], parts[1]);
-    let authReq: AuthReq;
+  private static getOAuthResult = async ({ acctEmail, scopes, save }: { acctEmail?: string, scopes: string[], save: boolean }): Promise<AuthRes> => {
```

Comment:
Method `getOAuthResult` should be called `renderOAuthWindowAndGetResult` to reflect what it actually does.